src/goto-programs/goto_trace.cpp

```diff
@@ -242,6 +242,15 @@ std::string trace_numeric_value(
       }
     }
   }
+  else if(expr.id() == ID_annotated_pointer_constant)
+  {
+    const auto &annotated_pointer_constant =
+      to_annotated_pointer_constant_expr(expr);
+    auto width = to_pointer_type(expr.type()).get_width();
+    auto value = annotated_pointer_constant.get_value();
+    auto c = constant_exprt(value, unsignedbv_typet(width));
```

Comment:
I'm yet to understand why this could be preferable over `constant_exprt c{value, ...};`? It's probably no measurably faster in terms of compile time, and will yield exactly the same code, but it just seems like noise in the program text?

Reply:
I guess you get a bit of symmetry between class types and the scalars.

Reply:
But then `int x{42};` is equally valid C++. Either way, I was just asking as I had seen it done by various people in different places.

Reply:
It's back to Basic!

Reply:
Maybe more seriously, the style also looks a bit more like a functional language, say think ocaml.